lib/ansible/modules/system/listen_ports_facts.py

```diff
@@ -0,0 +1,254 @@
+  sample:
+    - address: '0.0.0.0'
+      name: 'mysqld'
+      pid: 1223
+      port: 3306
+      protocol: 'tcp'
+      stime: 'Thu Feb  2 13:29:45 2017'
+      user: 'mysql'
+    - address: '0.0.0.0'
+      name: 'httpd'
+      pid: 905
+      port: 443
+      protocol: 'tcp'
+      stime: 'Thu Feb  2 13:29:45 2017'
+      user: 'apache'
+
+udp_listen_violations:
+  description: A list of processes that are listening on a UDP port that violated the whitelist_udp argument.
+  returned: when at least one process that is listening on a UDP port isn't included in the supplied UDP whitelist.
+  type: list
+  sample:
+    - address: '0.0.0.0'
+      name: 'rsyslogd'
+      pid: 609
+      port: 514
+      protocol: 'udp'
+      stime: 'Thu Feb  2 13:29:45 2017'
+      user: 'root'
+    - address: '0.0.0.0'
+      name: 'chrome'
+      pid: 4551
+      port: 5353
+      protocol: 'udp'
+      stime: 'Thu Feb  2 13:29:45 2017'
+      user: 'root'
+'''
+
+import re
+import platform
+from ansible.module_utils.basic import AnsibleModule
+
+
+def netStatParse(raw):
+    results = list()
+    for line in iter(raw.splitlines()):
+        listening_search = re.search('[^ ]+:[0-9]+', line)
+        if listening_search:
+            splitted = line.split()
+            conns = re.search('([^ ]+):([0-9]+)', splitted[3])
+            pidstr = ''
+            if 'tcp' in splitted[0]:
+                protocol = 'tcp'
+                pidstr = splitted[6]
+            elif 'udp' in splitted[0]:
+                protocol = 'udp'
+                pidstr = splitted[5]
+            pids = re.search('(([0-9]+)/(.*)|-)', pidstr)
+            if conns and pids:
+                address = conns.group(1)
+                port = conns.group(2)
+                if (pids.group(2)):
+                    pid = pids.group(2)
+                else:
+                    pid = 0
+                if (pids.group(3)):
+                    name = pids.group(3)
+                else:
+                    name = ''
+                result = dict(pid=int(pid), address=address, port=int(port), protocol=protocol, name=name)
+                if result not in results:
+                    results.append(result)
+            else:
+                raise EnvironmentError('Could not get process information for the listening ports.')
+    return results
+
+
+def applyWhitelist(portspids, whitelist=None):
+    if whitelist is None:
+        whitelist = list()
+    processes = list()
+    for p in portspids:
+        if int(p['port']) not in whitelist and str(p['port']) not in whitelist:
+            process = dict(pid=p['pid'], address=p['address'], port=p['port'], protocol=p['protocol'], name=p['name'], stime=p['stime'], user=p['user'])
+            if process not in processes:
+                processes.append(process)
+    return processes
+
+
+def main():
+
+    module = AnsibleModule(
+        argument_spec=dict(
+            whitelist_tcp=dict(required=False, type='list', default=list()),
```

Comment:
If it is not requires, don't add it.
```suggestion
            whitelist_tcp=dict(type='list', default=list()),
```